### ISSUE TITLE
mpas - error occurred while rendering a Vertical Cross Section Plot in VERDI

### ISSUE THREAD

**Issue by lizadams** [CEMPD/VERDI#121]
Dataset: history.2013-07-10.1step.nc
The above dataset was created from the full 24hour history file provided by Donna using the nco command:
ncea -F -d Time,1 history.2013-07-10.nc history.2013-07-10.1step.nc
Variable: smois (this occurs for any variable)
Exception in thread "AWT-EventQueue-0" java.lang.Error: Unresolved compilation problems: 
	The import ucar.ma2.ArrayLogFactory cannot be resolved
	ArrayLogFactory cannot be resolved

	at anl.verdi.plot.gui.MeshPlot.<init>(MeshPlot.java:114)
	at anl.verdi.plot.jfree.MPASXYBlockRenderer.<init>(MPASXYBlockRenderer.java:32)
	at anl.verdi.plot.types.VerticalCrossSectionPlot.createChart(VerticalCrossSectionPlot.java:398)
	at anl.verdi.plot.types.VerticalCrossSectionPlot.<init>(VerticalCrossSectionPlot.java:160)
	at anl.verdi.plot.types.VerticalCrossSectionPlot.<init>(VerticalCrossSectionPlot.java:106)
	at anl.verdi.plot.gui.PlotFactory.getVerticalCrossPlot(PlotFactory.java:109)
	at anl.verdi.plot.gui.VerticalCrossPlotCreator.createVertPlot(VerticalCrossPlotCreator.java:122)
	at anl.verdi.plot.gui.VerticalCrossPlotCreator.doCreatePlot(VerticalCrossPlotCreator.java:88)
	at anl.verdi.plot.gui.AbstractPlotCreator.createPlot(AbstractPlotCreator.java:39)
	at anl.verdi.plot.gui.action.VerticalCrossSection.actionPerformed(VerticalCrossSection.java:21)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

Header information for the dataset:
netcdf history.2013-07-10.1step {
dimensions:
        Time = UNLIMITED ; // (1 currently)
        nCells = 163842 ;
        nVertLevels = 50 ;
        nEdges = 491520 ;
        nVertices = 327680 ;
        TWO = 2 ;
        maxEdges = 10 ;
        maxEdges2 = 20 ;
        vertexDegree = 3 ;
        nVertLevelsP1 = 51 ;
        StrLen = 64 ;
        nOznLevels = 59 ;
        nMonths = 12 ;
        nSoilLevels = 4 ;
variables:
        float qv(Time, nCells, nVertLevels) ;
        float qc(Time, nCells, nVertLevels) ;
        float qr(Time, nCells, nVertLevels) ;
        float qi(Time, nCells, nVertLevels) ;
        float qs(Time, nCells, nVertLevels) ;
        float qg(Time, nCells, nVertLevels) ;
        float latCell(nCells) ;
        float lonCell(nCells) ;
        float xCell(nCells) ;
        float yCell(nCells) ;
        float zCell(nCells) ;
        int indexToCellID(nCells) ;
        float latEdge(nEdges) ;
        float lonEdge(nEdges) ;
        float xEdge(nEdges) ;
        float yEdge(nEdges) ;
        float zEdge(nEdges) ;
        int indexToEdgeID(nEdges) ;
        float latVertex(nVertices) ;
        float lonVertex(nVertices) ;
        float xVertex(nVertices) ;
        float yVertex(nVertices) ;
        float zVertex(nVertices) ;
        int indexToVertexID(nVertices) ;
        int cellsOnEdge(nEdges, TWO) ;
        int nEdgesOnCell(nCells) ;
        int nEdgesOnEdge(nEdges) ;
        int edgesOnCell(nCells, maxEdges) ;
        int edgesOnEdge(nEdges, maxEdges2) ;
        float weightsOnEdge(nEdges, maxEdges2) ;
        float dvEdge(nEdges) ;
        float dcEdge(nEdges) ;
        float angleEdge(nEdges) ;
        float areaCell(nCells) ;
        float areaTriangle(nVertices) ;
        int cellsOnCell(nCells, maxEdges) ;
        int verticesOnCell(nCells, maxEdges) ;
        int verticesOnEdge(nEdges, TWO) ;
        int edgesOnVertex(nVertices, vertexDegree) ;
        int cellsOnVertex(nVertices, vertexDegree) ;
        float kiteAreasOnVertex(nVertices, vertexDegree) ;
        float meshDensity(nCells) ;
        float zgrid(nCells, nVertLevelsP1) ;
        float fzm(nVertLevels) ;
        float fzp(nVertLevels) ;
        float zz(nCells, nVertLevelsP1) ;
        char xtime(Time, StrLen) ;
        float u(Time, nEdges, nVertLevels) ;
        float w(Time, nCells, nVertLevelsP1) ;
        float pressure(Time, nCells, nVertLevels) ;
        float surface_pressure(Time, nCells) ;
        float rho(Time, nCells, nVertLevels) ;
        float theta(Time, nCells, nVertLevels) ;
        float relhum(Time, nCells, nVertLevels) ;
        float divergence(Time, nCells, nVertLevels) ;
        float vorticity(Time, nVertices, nVertLevels) ;
        float ke(Time, nCells, nVertLevels) ;
        float uReconstructZonal(Time, nCells, nVertLevels) ;
        float uReconstructMeridional(Time, nCells, nVertLevels) ;
        int i_rainnc(Time, nCells) ;
        float rainnc(Time, nCells) ;
        float precipw(Time, nCells) ;
        float cuprec(Time, nCells) ;
        int i_rainc(Time, nCells) ;
        float rainc(Time, nCells) ;
        int kpbl(Time, nCells) ;
        float hpbl(Time, nCells) ;
        float hfx(Time, nCells) ;
        float qfx(Time, nCells) ;
        float cd(Time, nCells) ;
        float cda(Time, nCells) ;
        float ck(Time, nCells) ;
        float cka(Time, nCells) ;
        float lh(Time, nCells) ;
        float u10(Time, nCells) ;
        float v10(Time, nCells) ;
        float q2(Time, nCells) ;
        float t2m(Time, nCells) ;
        float th2m(Time, nCells) ;
        float gsw(Time, nCells) ;
        float glw(Time, nCells) ;
        float acsnow(Time, nCells) ;
        float pin(nOznLevels) ;
        float ozmixm(nCells, nOznLevels, nMonths) ;
        char mminlu(StrLen) ;
        float xland(Time, nCells) ;
        float ter(nCells) ;
        float skintemp(Time, nCells) ;
        float snow(Time, nCells) ;
        float snowh(Time, nCells) ;
        float sst(Time, nCells) ;
        float vegfra(Time, nCells) ;
        float xice(Time, nCells) ;
        float sh2o(Time, nCells, nSoilLevels) ;
        float smois(Time, nCells, nSoilLevels) ;
        float tslb(Time, nCells, nSoilLevels) ;

// global attributes:
                :model_name = "mpas" ;
                :core_name = "atmosphere" ;
                :source = "MPAS" ;
                :Conventions = "MPAS" ;
                :git_version = "unknown" ;
                :on_a_sphere = "YES" ;
                :sphere_radius = 6371229. ;
                :history = "Sun Oct 16 07:53:40 2016: ncea -F -d Time,1 history.2013-07-10.nc history.2013-07-10.1step.nc\n",
                        "mpirun -n 64 ./atmosphere_model" ;
                :parent_id = "" ;
                :mesh_spec = "" ;
                :config_time_integration = "SRK3" ;
                :config_dt = 150. ;
                :config_calendar_type = "gregorian" ;
                :config_start_time = "2013-07-01_00:00:00" ;
                :config_stop_time = "none" ;
                :config_run_duration = "31_00:00:00" ;
                :config_split_dynamics_transport = "YES" ;
                :config_number_of_sub_steps = 2 ;
                :config_dynamics_split_steps = 3 ;
                :config_h_mom_eddy_visc2 = 0. ;
                :config_h_mom_eddy_visc4 = 0. ;
                :config_v_mom_eddy_visc2 = 0. ;
                :config_h_theta_eddy_visc2 = 0. ;
                :config_h_theta_eddy_visc4 = 0. ;
                :config_v_theta_eddy_visc2 = 0. ;
                :config_horiz_mixing = "2d_smagorinsky" ;
                :config_len_disp = 25000. ;
                :config_visc4_2dsmag = 0.05 ;
                :config_del4u_div_factor = 1. ;
                :config_w_adv_order = 3 ;
                :config_theta_adv_order = 3 ;
                :config_scalar_adv_order = 3 ;
                :config_u_vadv_order = 3 ;
                :config_w_vadv_order = 3 ;
                :config_theta_vadv_order = 3 ;
                :config_scalar_vadv_order = 3 ;
                :config_scalar_advection = "YES" ;
                :config_positive_definite = "NO" ;
                :config_monotonic = "YES" ;
                :config_coef_3rd_order = 0.25 ;
                :config_mix_full = "YES" ;
                :config_epssm = 0.1 ;
                :config_smdiv = 0.1 ;
                :config_newpx = "NO" ;
                :config_apvm_upwinding = 0.5 ;
                :config_h_ScaleWithMesh = "YES" ;
                :config_num_halos = 2 ;
                :config_zd = 27000. ;
                :config_xnutr = 0.2 ;
                :config_restart_timestamp_name = "restart_timestamp" ;
                :config_pio_num_iotasks = 0 ;
                :config_pio_stride = 1 ;
                :config_block_decomp_file_prefix = "x4.163842.graph.info.part." ;
                :config_number_of_blocks = 0 ;
                :config_explicit_proc_decomp = "NO" ;
                :config_proc_decomp_file_prefix = "graph.info.part." ;
                :config_do_restart = "NO" ;
                :config_do_DAcycling = "NO" ;
                :config_print_global_minmax_vel = "YES" ;
                :config_print_global_minmax_sca = "NO" ;
                :input_soil_data = "STAS" ;
                :input_soil_temperature_lag = 140 ;
                :num_soil_layers = 4 ;
                :months = 12 ;
                :noznlev = 59 ;
                :naerlev = 29 ;
                :camdim1 = 4 ;
                :config_frac_seaice = "NO" ;
                :config_sfc_albedo = "YES" ;
                :config_sfc_snowalbedo = "YES" ;
                :config_sst_update = "YES" ;
                :config_sstdiurn_update = "NO" ;
                :config_deepsoiltemp_update = "NO" ;
                :config_o3climatology = "NO" ;
                :config_n_microp = 1 ;
                :config_radtlw_interval = "00:10:00" ;
                :config_radtsw_interval = "00:10:00" ;
                :config_conv_interval = "none" ;
                :config_pbl_interval = "none" ;
                :config_camrad_abs_update = "06:00:00" ;
                :config_greeness_update = "24:00:00" ;
                :config_bucket_update = "none" ;
                :config_physics_suite = "mesoscale_reference" ;
                :config_microp_scheme = "wsm6" ;
                :config_convection_scheme = "kain_fritsch" ;
                :config_fdda_scheme = "analysis" ;
                :config_fdda_int = 21600. ;
                :config_fdda_scale_min = 0. ;
                :config_fdda_scale_max = 100000. ;
                :config_fdda_t = "YES" ;
                :config_fdda_t_coef = 0.0003 ;
                :config_fdda_t_in_pbl = "NO" ;
                :config_fdda_t_min_layer = 0 ;
                :config_fdda_q = "YES" ;
                :config_fdda_q_coef = 3.e-05 ;
                :config_fdda_q_in_pbl = "NO" ;
                :config_fdda_q_min_layer = 0 ;
                :config_fdda_uv = "YES" ;
                :config_fdda_uv_coef = 0.0003 ;
                :config_fdda_uv_in_pbl = "NO" ;
                :config_fdda_uv_min_layer = 0 ;
                :config_lsm_scheme = "noah" ;
                :config_pbl_scheme = "ysu" ;
                :config_gwdo_scheme = "off" ;
                :config_radt_cld_scheme = "cld_fraction" ;
                :config_radt_lw_scheme = "rrtmg_lw" ;
                :config_radt_sw_scheme = "rrtmg_sw" ;
                :config_sfclayer_scheme = "monin_obukhov" ;
                :config_bucket_radt = 1000000000. ;
                :config_bucket_rainc = 100. ;
                :config_bucket_rainnc = 100. ;
                :file_id = "lnyq06ppi5" ;
                :NCO = "\"4.5.2\"" ;
                :nco_openmp_thread_number = 1 ;

**Comment by lizadams** [comment]
Compilation error in MeshPlot.java in this section of the code:
import ucar.ma2.ArrayLogFactory;   **The import ucar.ma2.ArrayLogFactory cannot be resolved**
	public DataFrame createLogDataFrame(DataFrame frame) {
		DataFrameBuilder builder = new DataFrameBuilder();
		builder.addDataset(dataset);
		builder.setVariable(frame.getVariable());
		builder.setArray(ArrayLogFactory.getArray(frame.getArray(), this.logBase));
		Axes<DataFrameAxis> axes = frame.getAxes();
		if (axes.getTimeAxis() != null)
			builder.addAxis(DataFrameAxis.createDataFrameAxis(axes.getTimeAxis(), axes.getTimeAxis().getArrayIndex()));
		if (axes.getZAxis() != null)
			builder.addAxis(DataFrameAxis.createDataFrameAxis(axes.getZAxis(), axes.getZAxis().getArrayIndex()));
		if (axes.getXAxis() != null)
			builder.addAxis(DataFrameAxis.createDataFrameAxis(axes.getXAxis(), axes.getXAxis().getArrayIndex()));
		if (axes.getYAxis() != null)
			builder.addAxis(DataFrameAxis.createDataFrameAxis(axes.getYAxis(), axes.getYAxis().getArrayIndex()));
		if (axes.getCellAxis() != null)
			builder.addAxis(DataFrameAxis.createDataFrameAxis(axes.getCellAxis(), axes.getCellAxis().getArrayIndex()));
		return builder.createDataFrame();
	}

**Comment by lizadams** [comment]
This was a problem when I rebuilt the netCDF java library to correct an issue with the Spherical Radius of the Earth Assumption for the WRF files.  I have now gone back to using the netCDF java that we built and provided with the VERDI 1.5.0_shapefile_mpas release.
This fixes it, but I need to copy all of the VERDI specific files that were changed under the verdi_data_loaders to the location on killdevil and rebuild so that we have both the WRF radius correction and all of the files needed by MPAS.
![verdi_data_loaders_file_for_netcdf_verdi_build](https://cloud.githubusercontent.com/assets/1183863/19417628/a42aa7ac-937f-11e6-8aaf-976ca737287e.png)
